components/02-elements/dropdown-list/dropdown-list-nested.js

```diff
@@ -0,0 +1,83 @@
+'use-strict'
+const dropdownItem      = document.querySelectorAll('.dropdown-list__item--parent .dropdown-list__label'),
+      dropdownInnerItem = document.querySelectorAll('.dropdown-list__inner-item--parent .dropdown-list__inner-label'),
+      activeClass       = 'open';
+
+function setListHeight(item) {
+  return Array.from(item.children)
+    .map(elem => elem.clientHeight)
+    .reduce((a, b) => a + b, 0);
+}
+
+function setAriaAttributes(label, content, expanded) {
+  if (expanded) {
+    label.setAttribute('aria-expanded', 'false')
+    content.setAttribute('aria-hidden', 'true');
+  }
+  else {
+    label.setAttribute('aria-expanded', 'true');
+    content.setAttribute('aria-hidden', 'false');
+  }
+}
+
+function toggleSubmenu(item, inner) {
+  const dropdownId = item.dataset.dropdown;
+  let dropdownList = document
+    .querySelector(`.dropdown-list__inner-list--level1[data-dropdown="${dropdownId}"]`),
+      innerLists   = item.parentNode.querySelectorAll('.dropdown-list__inner-list--level2'),
+      upperList    = null;
+
+  if (inner) {
+    dropdownList = item.parentNode.querySelector(`.dropdown-list__inner-list--level2[data-dropdown="${dropdownId}"]`);
+    innerLists = null;
+    upperList  = item.closest('.dropdown-list__inner-list--level1');
+  }
+
+  item.parentNode.classList.toggle('open');
+
+  if (dropdownList && dropdownList.clientHeight > 0) {
+    if (innerLists) {
+      innerLists.forEach(key => {
+        key.style.height = 0;
+        key.parentNode.classList.remove(activeClass);
+        const innerListLabel = key.parentNode.querySelector(`[data-dropdown="${key.getAttribute('id')}"]`);
+        setAriaAttributes(innerListLabel, key, true);
+      });
+    }
+    else if (upperList) {
+      upperList.style.height = upperList.clientHeight - dropdownList.clientHeight + 'px';
+    }
+
+    dropdownList.style.height = 0;
+    const  listLabel = dropdownList
+      .parentNode
+      .querySelector(`[data-dropdown="${dropdownList.getAttribute('id')}"]`);
+    setAriaAttributes(listLabel, dropdownList, true);
+  }
+  else if (dropdownList) {
+    const listHeight = setListHeight(dropdownList),
+          listLabel = dropdownList
+            .parentNode
+            .querySelector(`[data-dropdown="${dropdownList.getAttribute('id')}"]`);
+    dropdownList.style.height = listHeight + 'px';
+    setAriaAttributes(listLabel, dropdownList, false);
+
+    if (upperList) {
+      upperList.style.height = upperList.clientHeight + listHeight + 'px';
+    }
+  }
+}
+
+dropdownItem.forEach(
+  key => key.addEventListener('click', (e) => {
+    e.preventDefault();
+    toggleSubmenu(key, false);
+  }, false)
+);
+
+dropdownInnerItem.forEach(
+  key => key.addEventListener('click', (e) => {
+    e.preventDefault();
+    toggleSubmenu(key, true);
+  }, false)
+);
```

Comment:
I just have to say, that I don't like this code at all, but I have to live with it... 😢

Reply:
It hurts my heart... 😭  What I can improve here? make a `Class`? any suggestion? 

I made some small update for readability... And I have one suggestion - In both file I use 2 functions `setListHeight` and `setAriaAttributes`. I though we can use import here, but this element is needed for other people in project to be merge so maybe create an issue or something? What do you think?

Reply:
I just believe that it's possible to write a unified function, that can handle an unlimited number of levels, without hardcoding levels numbers, parents, nesting conditions etc. 

But I know it will take a lot of time to refactor it...

About the functions - ignore it, for now, that's just preview.